### PR TITLE
feat/microphone plugins + update logging and documentation

### PR DESCRIPTION
**Pull request by NeonDaniel**
Refactor to import from OPM
Add some logs to debug listener state and activity
Adds docstrings and comments
Includes some PEP formatting
Includes minor bugfixes/improvements

### CODE REVIEW

File: ovos_dinkum_listener/voice_loop/microphone.py [hunk 1]
```diff
@@ -59,27 +35,42 @@ class AlsaMicrophone(Microphone):
     _thread: Optional[Thread] = None
     _queue: "Queue[Optional[bytes]]" = field(default_factory=Queue)
     _is_running: bool = False
+    _lock = Lock()
```
Comment (JarbasAl) [inline ovos_dinkum_listener/voice_loop/microphone.py:38]: this class is also a plugin now, it + pyaudio microphone should be deleted

mic creation should use opm MicrophoneFactory to use a mic pllugin

Comment (JarbasAl) [reply ovos_dinkum_listener/voice_loop/microphone.py:38]: or for now, just add the plugin as a dependency and import alsa microphone from there

File: ovos_dinkum_listener/service.py [hunk 2]
```diff
@@ -135,18 +136,8 @@ def __init__(self, on_ready=on_ready, on_error=on_error,
...
-            sample_width=listener.get("sample_width", 2),
-            sample_channels=listener.get("sample_channels", 1),
-            chunk_size=listener.get("chunk_size", 4096),
-            period_size=listener.get("period_size", 1024),
-            multiplier=listener.get("multiplier", 1),
-            timeout=listener.get("audio_timeout", 5),
-            audio_retries=listener.get("audio_retries", 3),
-            audio_retry_delay=listener.get("audio_retry_delay", 1),
-        )
+        self.mic = mic or OVOSMicrophoneFactory.create(self.config)
```
Comment (JarbasAl) [inline ovos_dinkum_listener/service.py:141]: should we default to alsa in this config explicitly? configs in the wild will all be missing the default values for now

(and im not sure if alsa will be the default)

**Comment by NeonDaniel** [comment]
https://github.com/OpenVoiceOS/ovos-plugin-manager/pull/154

File: ovos_dinkum_listener/service.py [hunk 3]
```diff
@@ -135,18 +138,15 @@ def __init__(self, on_ready=on_ready, on_error=on_error,
...
-            chunk_size=listener.get("chunk_size", 4096),
-            period_size=listener.get("period_size", 1024),
-            multiplier=listener.get("multiplier", 1),
-            timeout=listener.get("audio_timeout", 5),
-            audio_retries=listener.get("audio_retries", 3),
-            audio_retry_delay=listener.get("audio_retry_delay", 1),
-        )
+        # Initialize with default (bundled) plugin
+        microphone_config = self.config.get("microphone", {})
+        microphone_config.setdefault('module', 'ovos-microphone-plugin-alsa')
```
Comment (JarbasAl) [inline ovos_dinkum_listener/service.py:143]: should use https://github.com/OpenVoiceOS/ovos-microphone-plugin-arecord if it's validated as working

pyaalsaaudio sometimes errors out installing, don't remember the details, either cython or swig or someother package needed at buildtime

arecord plugin has 0 python dependencies

Comment (NeonDaniel) [reply ovos_dinkum_listener/service.py:143]: Get the following exception when using arecord:
```
2023-05-22 11:54:00.752 - voice - ovos_dinkum_listener.service:run:210 - ERROR - Service failed to start
Traceback (most recent call last):
  File "/home/neon/venv/lib/python3.7/site-packages/ovos_dinkum_listener/service.py", line 201, in run
    self.voice_loop.run()
  File "/home/neon/venv/lib/python3.7/site-packages/ovos_dinkum_listener/voice_loop/voice_loop.py", line 171, in run
    assert chunk is not None, "No audio from microphone"
AssertionError: No audio from microphone
```

**Comment by NeonDaniel** [comment]
Validated with https://github.com/OpenVoiceOS/ovos-plugin-manager/pull/155

**Pull request merged**